src/Symfony/Component/Console/Output/OutputSectionReference.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Output;
+
+/**
+ * @internal
+ *
+ * @author Pierre du Plessis <pdples@gmail.com>
+ */
+final class OutputSectionReference
+{
+    private $ref = array();
+
+    public function addRef(ConsoleSectionOutput $output)
+    {
+        array_unshift($this->ref, $output);
+    }
+
+    public function calculateBuffer(ConsoleSectionOutput $output)
+    {
+        $buffer = 0;
+
+        foreach ($this->ref as $outputStream) {
+            if (spl_object_id($outputStream) === spl_object_id($output)) {
```

Comment:
just compare the objects themselves inside of comparing the ids explicitly. It will be easier to understand (and will avoid the dependency on the polyfill).